### ISSUE TITLE
Server Rejecting Same Version for Client

### ISSUE THREAD

**Issue by Gibil** [bernie-g/geckolib#459]
I have the exact same jar "geckolib-forge-1.20.1-4.2.1.jar" on both client and server, but the server rejects connections due to this mod not having the correct version.

The only mention of the mod in the server log is

```
[19Jul2023 14:57:21.918] [main/DEBUG] [net.minecraftforge.fml.loading.moddiscovery.ModFileParser/LOADING]: Considering mod file candidate /home/container/mods/geckolib-forge-1.20.1-4.2.1.jar
[19Jul2023 14:57:21.918] [main/DEBUG] [net.minecraftforge.fml.loading.moddiscovery.ModFileInfo/LOADING]: Found valid mod file geckolib-forge-1.20.1-4.2.1.jar with {geckolib} mods - versions {4.2.1}

[19Jul2023 14:57:22.149] [main/DEBUG] [net.minecraftforge.fml.loading.moddiscovery.AbstractJarFileDependencyLocator/]: Failed to load resource META-INF/jarjar/metadata.json from geckolib-forge-1.20.1-4.2.1.jar, it does not contain dependency information.

 [19Jul2023 14:57:22.282] [main/WARN] [net.minecraftforge.jarjar.selection.JarSelector/]: Attempted to select a dependency jar for JarJar which was passed in as source: geckolib. Using Mod File: /home/container/mods/geckolib-forge-1.20.1-4.2.1.jar
```

![image](https://github.com/bernie-g/geckolib/assets/1112830/2b44c994-80fd-43da-aa7b-2a67baed34b2)

Forge 47.1.3

**Comment by Tslat** [comment]
Provide your full client and server log files

**Comment by Gibil** [comment]
I apologize, but after deleting the modpack in the client and reinstalling, and then loading the backup of the world prior to updating the server (and deleting config) I am not able to reproduce.  While geckolib was deleted in the pack's mods/ folder it somehow still showed up in the loaded modlist in-game. I can't begin to guess as to how.  Unless modlist is cached somehow... Logs are attached from when it was occurring, but I'm guessing this should be closed.

[debug.log](https://github.com/bernie-g/geckolib/files/12100691/debug.log)
[latest.log](https://github.com/bernie-g/geckolib/files/12100714/latest.log)

**Comment by Tslat** [comment]
Yeah with no reproduction we can't say this is anything more than a random conflict or glitch with caching or something

We'll see what happens I guess

**Comment by Gibil** [comment]
Apparently it does eventually come back... Modpack version and server haven't changed. Once it starts then all users experience the same thing. I did notice an apparently vanilla connection attempt, but not sure that's related.

[server - debug.log](https://github.com/bernie-g/geckolib/files/12109578/server.-.debug.log)
[client-debug.log](https://github.com/bernie-g/geckolib/files/12109579/client-debug.log)

[pack mod list](https://pastebin.com/Vw53bSV5)
[server mod list](https://pastebin.com/Y891H0cw)

I'll do anything I can to dig further.

**Comment by Gibil** [comment]
Finally figured something useful out.
Ars Nouveau is also a dependent on Geckolib, even though CurseForge does not list that... It had a recent update with "Geckolib is now JIJ instead of shaded" in the changelog. I don't know what that means, but with Ars 4.2.0-all the Geckolib version is rejected by the server even when both server and client have the same geckolib jar as I mentioned before, showing up as this in the client debug log:

[Netty Client IO #2/DEBUG] [net.minecraftforge.network.NetworkRegistry/NETREGISTRY]: Channel 'geckolib:main' : Version test of 'ABSENT ?' from server : REJECTED

This does not occur, and my issue seems to be fixed 🤞, when downgrading Ars to 4.1.1.  Can a mod load its dependencies without the client installing them? I am completely unknowledgeable of such things.  But Occultism also depends on Geckolib so the separate geckolib jar does need to be there... Anyway thanks. @Tslat
